Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -512,7 +552,16 @@ public void WaitForTasksCompletion(IEnumerable<string> taskIds)
       }
     }
 
-    return ms.ToArray();
+    var res = new byte[chunks.Sum(rm => rm.Length)];
```

Comment:
You can compute the sum without iterating again over chunks, by computing it by hand during the previous loop.